src/components/Notifications/TimeConfig.tsx

```diff
@@ -126,10 +133,113 @@ export const TimeConfigComponent: React.FunctionComponent = () => {
                 }
             });
         }
-    }, [ addDangerNotification, addSuccessNotification, saveTimePreference.mutate, timeSelect ]);
+
+        if (notificationsOverhaul) {
+            setIsModalOpen(false);
+        }
+    }, [ addDangerNotification, addSuccessNotification, saveTimePreference.mutate, timeSelect, notificationsOverhaul ]);
 
     const isLoading = saveTimePreference.loading || getTimePreference.loading;
 
+    if (notificationsOverhaul) {
+        const handleModalToggle = (_event: KeyboardEvent | React.MouseEvent) => {
+            setIsModalOpen(!isModalOpen);
+        };
+
+        return (
+            <>
+                <p>
+                    <OutlinedClockIcon color='var(--pf-v5-global--palette--cyan-200)' />
+                    &nbsp;{`Any daily digest emails you've opted into will be sent at ${timePref ? timePref : '00:00'} UTC`}
+                </p>
+                <Button variant="link" onClick={ handleModalToggle } ouiaId="TimeConfigModal">
+                    Edit time settings
+                </Button>
+                <Modal
+                    variant={ ModalVariant.small }
+                    isOpen={ isModalOpen }
+                    onClose={ () => handleModalToggle }
```

Comment:
This will probably not work, you have to assign the callback itself.
```suggestion
                    onClose={ handleModalToggle }
```

Reply:
Fixed. Patternfly v4's onClose doesn't accept EventHandler